### PR TITLE
Move filter semantics out of syntax section

### PR DESCRIPTION
**Pull request by glyn**
Fixes https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/issues/208

### CODE REVIEW

File: draft-ietf-jsonpath-base.md [hunk 1]
```diff
@@ -1224,6 +1212,11 @@ The `filter-selector` works with arrays and objects exclusively. Its result is a
 A relative path, beginning with `@`, refers to the current array element or member value as the
 filter selector iterates over the array or object.
 
+A member or element value by itself in a Boolean context is
```
Comment (cabo) [inline draft-ietf-jsonpath-base.md:1215]: That doesn't work for me.
Once something is a value, it is that value; no further conversion happens.
This sentence was originally about queries that deliver a node list.
A value that doesn't exist isn't a value, no?

Comment (glyn) [reply draft-ietf-jsonpath-base.md]: Yes, I've added a commit with better wording.

**Review by timbray — APPROVED** [review]

File: draft-ietf-jsonpath-base.md [hunk 2]
```diff
@@ -1253,6 +1244,13 @@ the comparison is between numeric values which satisfy the comparison.
 Note that comparisons between structured values, even if the values are equal, produce a "false" comparison result.
 <!-- issue: comparison with structured value -->
 
+Data types are not implicitly converted in comparisons.
+So `"13 == '13'"` selects no node.
+
+Regular expression tests can be applied to JSON string values
+({{Section 7 of -json}}) only
+(on the left-hand side of `=~`); they yield false otherwise.
```
[the remark this answers is not in the document]
Comment (glyn) [reply draft-ietf-jsonpath-base.md]: Yes, that's better thanks. Adopted.

I also like the phrase "yields true" and have adopted it throughout for results of boolean expressions.

File: draft-ietf-jsonpath-base.md [hunk 3]
```diff
@@ -1234,30 +1222,39 @@ The `filter-selector` works with arrays and objects exclusively. Its result is a
 A relative path, beginning with `@`, refers to the current array element or member value as the
 filter selector iterates over the array or object.
 
-When a path resulting in an empty nodelist appears on either side of a comparison, the result of the comparison is
+A singular path by itself in a Boolean context is an existence test which yields true if the path selects a node and yields false if the path does not select a node.
```
Comment (cabo) [inline draft-ietf-jsonpath-base.md:1225]: Did we ever clarify that we don't want existence tests on general paths?

File: draft-ietf-jsonpath-base.md [hunk 4]
```diff
@@ -1234,30 +1222,39 @@ The `filter-selector` works with arrays and objects exclusively. Its result is a
...
+To be more specific about the actual value of a node selected by a path, explicit comparisons are necessary. This existence test — as an exception to the general rule — also works with nodes with structured values.
```
Comment (cabo) [inline draft-ietf-jsonpath-base.md:1226]: s/be more specific about the actual/make use of the JSON/

File: draft-ietf-jsonpath-base.md [hunk 5]
```diff
@@ -1234,30 +1222,39 @@ The `filter-selector` works with arrays and objects exclusively. Its result is a
...
+
+When a path resulting in an empty nodelist appears on either side of a comparison, the comparison yields
 true if and only if the comparison operator is `==`, `>=` or `<=` and the other side of the comparison is also a path
```
Comment (cabo) [inline draft-ietf-jsonpath-base.md:1229]: So != is always false, while == can be true?
Doesn't look right.

**Review by cabo — COMMENTED** [review]
I'm not sure we are done with this yet.  See comments.

File: draft-ietf-jsonpath-base.md (hunk 3 above)
Comment (glyn) [reply draft-ietf-jsonpath-base.md:1225]: I was under the impression that the consensus was to restrict existence tests to singular paths. That seems to be clear in the spec. Did you want further clarification in the spec or to revisit the question?

File: draft-ietf-jsonpath-base.md (hunk 4 above)
Comment (glyn) [reply draft-ietf-jsonpath-base.md:1226]: Yes, I'll put that in a new PR, thanks.

File: draft-ietf-jsonpath-base.md (hunk 5 above)
Comment (glyn) [reply draft-ietf-jsonpath-base.md:1229]: I tried fixing this in https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/pull/224.

File: draft-ietf-jsonpath-base.md (hunk 4 above)
Comment (glyn) [reply draft-ietf-jsonpath-base.md:1226]: Please see https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/pull/224.

**Pull request merged**